modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java

```diff
@@ -159,35 +162,90 @@ private boolean mediate(MessageContext synCtx, String format) {
         return true;
     }
 
-    /**
-     * Calls the replace function. isFormatDynamic check is used to remove indentations which come from registry based
-     * configurations.
-     * @param result
-     * @param synCtx
-     * @param format
-     */
-    private void regexTransform(StringBuffer result, MessageContext synCtx, String format) {
-        if (isFormatDynamic()) {
-            String key = formatKey.evaluateValue(synCtx);
-            Object entry = synCtx.getEntry(key);
-            if(entry == null){
-            	handleException("Key " + key + " not found ", synCtx);
-            }            
-            String text = "";
-            if (entry instanceof OMElement) {
-                OMElement e = (OMElement) entry;
-                removeIndentations(e);
-                text = e.toString();
-            } else if (entry instanceof OMText) {
-                text =  ((OMText) entry).getText();
-            } else if (entry instanceof String) {
-                text = (String) entry;
-            }
-            replace(text, result, synCtx);
-        } else {
-            replace(format, result, synCtx);
-        }
-    }
+	/**
+	 * Calls the replaceCTX function. isFormatDynamic check is used to remove indentations which come from registry based
+	 * configurations.
+	 * @param resultCTX
+	 * @param synCtx
+	 * @param format
+	 */
+	private void regexTransformCTX(StringBuffer resultCTX, MessageContext synCtx, String format) {
+		if (isFormatDynamic()) {
+			String key = formatKey.evaluateValue(synCtx);
+			Object entry = synCtx.getEntry(key);
+			if(entry == null){
+				handleException("Key " + key + " not found ", synCtx);
+			}
+			String text = "";
+			if (entry instanceof OMElement) {
+				OMElement e = (OMElement) entry;
+				removeIndentations(e);
+				text = e.toString();
+			} else if (entry instanceof OMText) {
+				text =  ((OMText) entry).getText();
+			} else if (entry instanceof String) {
+				text = (String) entry;
+			}
+			replaceCTX(text, resultCTX, synCtx);
+		} else {
+			replaceCTX(format, resultCTX, synCtx);
+		}
+	}
+
+	/**
+	 * Replaces the payload format with property values from messageContext.
+	 *
+	 * @param format
+	 * @param resultCTX
+	 * @param synCtx
+	 */
+	private void replaceCTX(String format, StringBuffer resultCTX, MessageContext synCtx) {
+		Matcher ctxMatcher;
+
+		if (mediaType != null && (mediaType.equals(JSON_TYPE) || mediaType.equals(TEXT_TYPE))) {
+			ctxMatcher=ctxPattern.matcher(format);
+		} else {
+			ctxMatcher=ctxPattern.matcher("<pfPadding>" + format + "</pfPadding>");
+		}
+		while (ctxMatcher.find()) {
+			String ctxMatchSeq = ctxMatcher.group();
+			String expressionTxt = ctxMatchSeq.substring(5, ctxMatchSeq.length());
+
+			String replaceValue = synCtx.getProperty(expressionTxt).toString();
+
+			if(mediaType.equals(JSON_TYPE) && inferReplacementType(replaceValue).equals(XML_TYPE)) {
```

Comment:
Could have been better to have as if(JSON_TYPE.equals(mediaType)) to avoid possible NPEs. Also check formatting. Check whether your IDE is configured with the wso2 code template.